src/title.rs

```diff
@@ -0,0 +1,55 @@
+use std::env;
+use std::fs::read_to_string;
+
+use crate::config::Config;
+use crate::Module;
+
+const HOSTNAME_PATH: &str = "/etc/hostname";
+
+pub struct Title {
+    user: Option<String>,
+    hostname: Option<String>,
+}
+
+fn get_user() -> Option<String> {
+    match env::var("USER") {
+        Ok(user) => Some(user),
+        Err(e) => {
+            eprintln!("Failed to get user, {}", e);
+            None
+        }
+    }
+}
+
+fn get_hostname() -> Option<String> {
+    match read_to_string(HOSTNAME_PATH) {
+        Ok(hostname) => Some(hostname),
+        Err(e) => {
+            eprintln!("Failed to get hostname, {}", e);
+            None
+        }
+    }
+}
```

Comment:
i've missed this before but since these functions are very simple we can just have them inlined in the constructor